### ISSUE TITLE
Supporting ResNet 1by2 variants

### ISSUE THREAD

**Issue by avitex** [keras-team/keras-applications#115]
A basic way to implement this would be simply defining ResNet50_1by2 variants like in [this commit](https://github.com/avitex/keras-applications/commit/b41d05580a6bbf6b1b03f08f1a0d6451bdc2dba0). However this would break the convention of the provided ImageNet weights (unless they were generated of course). Instead would propose a generic `filter_mult` arg which would be used like so:

```py
def ResNet50(include_top=True,
             weights='imagenet',
             input_tensor=None,
             input_shape=None,
             pooling=None,
             classes=1000,
             filter_mult=1,
             **kwargs):
    def stack_fn(x, filter_mult):
        x = stack1(x, int(64 * filter_mult), 3, stride1=1, name='conv2')
        x = stack1(x, int(128 * filter_mult), 4, name='conv3')
        x = stack1(x, int(256 * filter_mult), 6, name='conv4')
        x = stack1(x, int(512 * filter_mult), 3, name='conv5')
        return x
    return ResNet(stack_fn, False, True, 'resnet50',
                  include_top, weights,
                  input_tensor, input_shape,
                  pooling, classes, filter_mult,
                  **kwargs)
```

With any value other than `1`, an error would be thrown if weights still specified 'imagenet':

```py
if weights == 'imagenet' and filter_mult != 1:
    # throw error
```

Happy to submit a PR for this :)

**Comment by 2sang** [comment]
Though it sounds completely fair to me to have more small variants of the model, subclassed models should be restricted to the more widely known ones that people agree with, IMHO. 
Or, we can have them as parameterized form when the option for the variants benefit the model hugely. :)

**Comment by avitex** [comment]
> I think subclassed models in this project should be restricted to the more widely known ones that people agree with, In my humble opinion.

I completely agree, it would be a little optimistic to assume every model used ever would be present, and the line has been drawn somewhere
